tests/cloner_test.go

```diff
@@ -640,6 +738,260 @@ var _ = Describe("all clone tests", func() {
+				dataVolume.Spec.Storage.VolumeMode = &volumeMode
+				dataVolume, err := utils.CreateDataVolumeFromDefinition(f.CdiClient, f.Namespace.Name, dataVolume)
+				Expect(err).ToNot(HaveOccurred())
+				f.ForceBindPvcIfDvIsWaitForFirstConsumer(dataVolume)
+				sourcePvc, err := f.K8sClient.CoreV1().PersistentVolumeClaims(dataVolume.Namespace).Get(context.TODO(), dataVolume.Name, metav1.GetOptions{})
+				Expect(err).ToNot(HaveOccurred())
+
+				SetFilesystemOverhead(f, targetOverHead, targetOverHead)
+				targetDV := utils.NewDataVolumeForCloningWithEmptySize("target-dv", sourcePvc.Namespace, sourcePvc.Name, nil, &volumeMode)
+				controller.AddAnnotation(targetDV, controller.AnnDeleteAfterCompletion, "false")
+				targetDataVolume, err := utils.CreateDataVolumeFromDefinition(f.CdiClient, f.Namespace.Name, targetDV)
+				Expect(err).ToNot(HaveOccurred())
+				targetPvc, err := utils.WaitForPVC(f.K8sClient, targetDataVolume.Namespace, targetDataVolume.Name)
+				Expect(err).ToNot(HaveOccurred())
+				f.ForceBindPvcIfDvIsWaitForFirstConsumer(targetDataVolume)
+
+				By("Wait for target PVC Bound phase")
+				err = utils.WaitForPersistentVolumeClaimPhase(f.K8sClient, f.Namespace.Name, v1.ClaimBound, targetPvc.Name)
+				Expect(err).ToNot(HaveOccurred())
+				By("Wait for target DV Succeeded phase")
+				err = utils.WaitForDataVolumePhaseWithTimeout(f, f.Namespace.Name, cdiv1.Succeeded, "target-dv", cloneCompleteTimeout)
+				Expect(err).ToNot(HaveOccurred())
+
+				// Compare the two clones to see if they have the same hash
+				compareCloneWithSource(sourcePvc, targetPvc, diskImagePath, diskImagePath)
+				deleteAndWaitForVerifierPod()
+				// Check if the PermissiveClone annotation exists in target PVC
+				By("Check expected annotations")
+				_, available := targetPvc.Annotations[controller.AnnPermissiveClone]
+				Expect(available).To(Equal(true))
+			},
+				Entry("Smaller overhead in source than in target", "0.50", "0.30"),
+				Entry("Bigger overhead in source than in target", "0.30", "0.50"),
+			)
+
+			It("Should only use size-detection pod when cloning a PVC for the first time", func() {
+				dataVolume := utils.NewDataVolumeWithHTTPImportAndStorageSpec(dataVolumeName, "200Mi", fmt.Sprintf(utils.TinyCoreIsoURL, f.CdiInstallNs))
+				dataVolume.Spec.Storage.VolumeMode = &volumeMode
+				dataVolume, err := utils.CreateDataVolumeFromDefinition(f.CdiClient, f.Namespace.Name, dataVolume)
+				Expect(err).ToNot(HaveOccurred())
+				f.ForceBindPvcIfDvIsWaitForFirstConsumer(dataVolume)
+				sourcePvc, err := f.K8sClient.CoreV1().PersistentVolumeClaims(dataVolume.Namespace).Get(context.TODO(), dataVolume.Name, metav1.GetOptions{})
+				Expect(err).ToNot(HaveOccurred())
+
+				By("Wait for source DV Succeeded phase")
+				err = utils.WaitForDataVolumePhaseWithTimeout(f, f.Namespace.Name, cdiv1.Succeeded, dataVolumeName, cloneCompleteTimeout)
+				Expect(err).ToNot(HaveOccurred())
+
+				wg.Add(1)
```

Comment:
Right but doing the thread doesn't really much either, because it is entirely possible that the pod is created, finishes and is removed inside the 1 second refresh of the eventually, so there is no guarantee that your thread will catch the pod either. It might make sense to respect the `AnnPodRetainAfterCompletion` on the datavolume. Basically if that annotation is on the DV, then any pods related to it being populated should not get deleted after they complete. Then you have plenty of time to inspect the pod and see if it did what you wanted.

Reply:
You are right, I somehow missed that. I'll try that annotation, thanks!